game-lobby/src/main/kotlin/GameLobby.kt

```diff
@@ -0,0 +1,199 @@
+                ).aggregate(
+                    { AllOpenGames() },
+                    { _, v, allGames ->
+                        allGames.execute(
+                            jsonMapper.readValue(v, GameCommand::class.java)
+                        )
+                        allGames
+                    },
+                    Materialized.`as`<String, AllOpenGames, KeyValueStore<Bytes, ByteArray>>(
+                        Topics.OPEN_GAMES_STORE_NAME_LOCAL
+                    ).withKeySerde(
+                        Serdes.String()
+                    ).withValueSerde(
+                        Serdes.serdeFrom(
+                            AllOpenGamesSerializer(),
+                            AllOpenGamesDeserializer()
+                        )
+                    )
+                )
+
+        aggregateAll.toStream()
+            .map { k, v ->
+                val json = jsonMapper.writeValueAsString(v)
+                println("Aggregated $json")
+                KeyValue(k, json)
+            }.to(Topics.OPEN_GAMES, Produced.with(Serdes.String(), Serdes.String()))
+
+        // expose the aggregated as a global ktable
+        // so that we can join against it
+        val allOpenGames: GlobalKTable<String, AllOpenGames> =
+            streamsBuilder.globalTable(
+                Topics.OPEN_GAMES,
+                Materialized.`as`<String, AllOpenGames, KeyValueStore<Bytes, ByteArray>>
+                    (Topics.OPEN_GAMES_STORE_NAME_GLOBAL)
+                    .withKeySerde(Serdes.String())
+                    .withValueSerde(
+                        Serdes.serdeFrom(
+                            AllOpenGamesSerializer(),
+                            AllOpenGamesDeserializer()
+                        )
+                    )
+            )
+
+        val findPublicGameStream: KStream<ReqId, FindPublicGame> =
+            streamsBuilder.stream<ReqId, String>(
+                Topics.FIND_PUBLIC_GAME,
+                Consumed.with(Serdes.UUID(), Serdes.String())
+            )
+                .mapValues { v -> jsonMapper.readValue(v, FindPublicGame::class.java) }
+
+        val fpgKeyJoiner: KeyValueMapper<ReqId, FindPublicGame, String> =
+            KeyValueMapper { _: ReqId, // left key
+                             _: FindPublicGame ->
+                // left value
+
+                // use a trivial join, so that all queries are routed to the same store
+                AllOpenGames.TOPIC_KEY
+            }
+
+        val fpgValueJoiner: ValueJoiner<FindPublicGame, AllOpenGames, FindPublicGameAllOpenGames> =
+            ValueJoiner { leftValue:
+                          FindPublicGame,
+                          rightValue:
+                          AllOpenGames ->
+                FindPublicGameAllOpenGames(leftValue, rightValue)
+            }
+
+        val fpgJoinAllOpenGames =
+            findPublicGameStream.leftJoin(allOpenGames, fpgKeyJoiner, fpgValueJoiner)
+
+        val fpgBranches =
```

Comment:
https://docs.confluent.io/current/streams/developer-guide/dsl-api.html

Add a predicate for "no match"